vm/systemSmartContracts/staking.go

```diff
@@ -565,12 +565,15 @@ func (s *stakingSC) unStakeAtEndOfEpoch(args *vmcommon.ContractCallInput) vmcomm
 		return vmcommon.Ok
 	}
 
-	if !registrationData.Staked {
-		log.Debug("stakingSC.unStakeAtEndOfEpoch: cannot unStake node which was already unStaked")
+	if !registrationData.Staked && !registrationData.Waiting {
+		log.Debug("stakingSC.unStakeAtEndOfEpoch: cannot unStake node which was already unStaked", "blsKey", hex.EncodeToString(args.Arguments[0]))
 		return vmcommon.Ok
 	}
 
-	s.removeFromStakedNodes()
+	if registrationData.Staked {
```

Comment:
could this method call result in too many nodes leaving? the unstakeMethod checks:

```
	if !s.canUnStake() {
		s.eei.AddReturnMessage("unStake is not possible as too many left")
		return vmcommon.UserError
	}
```

Reply:
here you unStake all - but if you reach this point where cannot unstake - the system will toggle pause.